tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/test_formula_5_4.py

```diff
@@ -0,0 +1,97 @@
+        manually_calculated_result = 0.015  # kN
+
+        assert form_5_4 == pytest.approx(expected=manually_calculated_result, rel=1e-4)
+
+    def test_raise_error_when_negative_theta_i_is_given(self) -> None:
+        """Test a negative value for theta_i."""
+        # Example values
+        theta_i = -0.003
+        n_a = 5
+        n_b = 10
+
+        with pytest.raises(NegativeValueError):
+            Form5Dot4TTransverseForceEffectBracingSystem(theta_i=theta_i, n_a=n_a, n_b=n_b)
+
+    def test_raise_error_when_negative_n_a_is_given(self) -> None:
+        """Test a negative value for n_a."""
+        # Example values
+        theta_i = 0.003
+        n_a = -5
+        n_b = 10
+
+        with pytest.raises(NegativeValueError):
+            Form5Dot4TTransverseForceEffectBracingSystem(theta_i=theta_i, n_a=n_a, n_b=n_b)
+
+    def test_raise_error_when_negative_n_b_is_given(self) -> None:
+        """Test a negative value for n_b."""
+        # Example values
+        theta_i = 0.003
+        n_a = 5
+        n_b = -10
+
+        with pytest.raises(NegativeValueError):
+            Form5Dot4TTransverseForceEffectBracingSystem(theta_i=theta_i, n_a=n_a, n_b=n_b)
+
+    def test_latex(self) -> None:
+        """Test the latex representation of the formula."""
+        # Example values
+        theta_i = 0.003  # -
+        n_a = 5  # kN
+        n_b = 10  # kN
+
+        # Object to test
+        form_5_4_latex = Form5Dot4TTransverseForceEffectBracingSystem(
+            theta_i=theta_i,
+            n_a=n_a,
+            n_b=n_b,
+        ).latex()
+
+        # Expected result
+        latex_complete = r"H_{i} = Θ_{i} \cdot (N_{b} - N_{a}) = 0.003 \cdot (10.000 - 5.000) = 0.015"
+        latex_short = r"H_{i} = 0.015"
+
+        latex_test_sample = {
+            "complete": {
+                "expected": latex_complete,
+                "actual": form_5_4_latex.complete,
+            },
+            "short": {"expected": latex_short, "actual": form_5_4_latex.short},
+            "string": {"expected": latex_complete, "actual": str(form_5_4_latex)},
+        }
+
+        assertion_errors = []
+
+        # A try-except block inside a loop is considered bad practice by Ruff
+        # However, the documentation states that ignoring this will only have a negligible impact on performance
+        # https://docs.astral.sh/ruff/rules/try-except-in-loop/
+        # In this case, the desired behavior is to collect all the errors and raise them at once
+        for representation, values in latex_test_sample.items():
+            try:
+                assert values["expected"] == values["actual"]
+            except AssertionError as error_message:  # noqa: PERF203
+                assertion_errors.append(rf"Error in {representation} representation. {error_message}")
+
+        if assertion_errors:
+            raise AssertionError("{} errors occurred:\n{}".format(len(assertion_errors), "\n".join(assertion_errors)))
```

Comment:
This implementation is inspired by subTest (https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests). Pytest does not have an equivalent for subTest. I have improved this code by using parametrize (https://docs.pytest.org/en/7.1.x/how-to/parametrize.html). I suggest you do that as well! It's much cleaner. Here's how it would look like:

```suggestion
    @pytest.mark.parametrize(
        ("representation", "expected"),
        [
            (
                "complete",
                r"H_{i} = Θ_{i} \cdot (N_{b} - N_{a}) = 0.003 \cdot (10.000 - 5.000) = 0.015",
            ),
            ("short", r"H_{i} = 0.015"),
            (
                "string",
                r"H_{i} = Θ_{i} \cdot (N_{b} - N_{a}) = 0.003 \cdot (10.000 - 5.000) = 0.015",
            ),
        ],
    )
    def test_latex(self, representation: str, expected: str) -> None:
        """Test the latex representation of the formula."""
        # Example values
        theta_i = 0.003  # -
        n_a = 5  # kN
        n_b = 10  # kN

        # Object to test
        form_5_4_latex = Form5Dot4TTransverseForceEffectBracingSystem(
            theta_i=theta_i,
            n_a=n_a,
            n_b=n_b,
        ).latex()

        actual = {
            "complete": form_5_4_latex.complete,
            "short": form_5_4_latex.short,
            "string": str(form_5_4_latex),
        }

        assert (
            actual[representation] == expected
        ), f"{representation} representation failed."
```

Reply:
Great suggestion! Thank you :-)